### PR TITLE
Merge Recent Works on Documentation

### PR DESCRIPTION
**Pull request by khajavi**
This pull request contains the following works on documentation:
- ZIO 2.0 Migration Guide
- Metric Data Types
- Reorganization of data types

It would be great to see them on the microsite, to see whether any problems exist.

### CODE REVIEW

**Comment by atooni** [comment]
Hi @khajavi, 

I have just tested your latest work for the documentation and noticed there was glitch creating the site. I wanted to create a PR against your PR, so that you can adopt the changes I have made, so that the site would build again. 

Have you disabled PRs in your fork ?

**Comment by khajavi** [comment]
> Hi @khajavi,
> 
> I have just tested your latest work for the documentation and noticed there was glitch creating the site. I wanted to create a PR against your PR, so that you can adopt the changes I have made, so that the site would build again.
> 
> Have you disabled PRs in your fork ?

Thank you for your help.
I didn't disable it. Can't you send PR to my PR?

**Comment by atooni** [comment]
> > Hi @khajavi,
> > I have just tested your latest work for the documentation and noticed there was glitch creating the site. I wanted to create a PR against your PR, so that you can adopt the changes I have made, so that the site would build again.
> > Have you disabled PRs in your fork ?
> 
> Thank you for your help. I didn't disable it. Can't you send PR to my PR?

No - just tried - let me try again

**Comment by atooni** [comment]
Hi again, 

github doesnt let me create a PR to your changes. Perhaps you can pull them yourself by creating an additional remote and merge my changes. My changes are here : https://github.com/blended-zio/zio/tree/docs-2.x

**Comment by adamgfraser** [comment]
@khajavi @atooni Great to get this documentation out! Let me know when this is ready to go.

**Comment by khajavi** [comment]
> Hi again,
> 
> github doesnt let me create a PR to your changes. Perhaps you can pull them yourself by creating an additional remote and merge my changes. My changes are here : https://github.com/blended-zio/zio/tree/docs-2.x

Hi Andreas,
I appreciate your effort. I have merged and pushed your change now.

**Comment by khajavi** [comment]
> @khajavi @atooni Great to get this documentation out! Let me know when this is ready to go.
All tests are passing. It's ready to go after Andreas's confirmation. /cc @atooni

**Comment by atooni** [comment]
@adamgfraser @khajavi  Looking good from my end
